### PR TITLE
Only enqueue non-terminal tasks

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
I find that Propeller tries to process the completed task from the `workqueue` because we enqueue the completed item. we should check the item's status before we enqueue them.

Problem:

| Time  | WorkQueue | [Worker](https://github.com/flyteorg/flytestdlib/blob/a8f2b07cd8434b11d9aef9ff7e6d04f7ded4f838/cache/auto_refresh.go#L156-L162) | node Evaluation | 
| ------------- | ------------ |------------- | -----------|
| T1  | Add item1  |                   |      |
| T2  |   |                Process item1  (job finish)       |        |
| T3  | Add item1  |        |         |
| T4  |  |               Process item1  again        |        |
| T5  |  |                       |   Update the node status (running -> succeed)      |

After:

| Time  | WorkQueue | [Worker](https://github.com/flyteorg/flytestdlib/blob/a8f2b07cd8434b11d9aef9ff7e6d04f7ded4f838/cache/auto_refresh.go#L156-L162) | node Evaluation | 
| ------------- | ------------ |------------- | -----------|
| T1  | Add item1  |                   |      |
| T2  |   |                Process item1  (job finish)       |        |
| T3  |  |        |         |
| T4  |  |                     |        |
| T5  |  |                       |   Update the node status (running -> succeed)      |

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: cache/auto_refresh.go [hunk 1]
```diff
@@ -209,9 +211,14 @@ func (w *autoRefresh) enqueueBatches(ctx context.Context) error {
...
 	for _, k := range keys {
+		if w.toDelete.Contains(k) {
+			w.lruMap.Remove(k)
+			w.toDelete.Remove(k)
+			continue
+		}
 		// If not ok, it means evicted between the item was evicted between getting the keys and this update loop
 		// which is fine, we can just ignore.
-		if value, ok := w.lruMap.Peek(k); ok && !w.toDelete.Contains(k) {
+		if value, ok := w.lruMap.Peek(k); ok && !value.(Item).IsTerminal() {
```
Comment (kumare3) [inline cache/auto_refresh.go:221]: should we check that the type of value is indeed Item?
if item, ok := value.(Item); ok {
}

Comment (pingsutw) [reply cache/auto_refresh.go]: updated it

**Review by EngHabu — APPROVED** [review]
